nexus/db-queries/src/transaction_retry.rs

```diff
@@ -0,0 +1,141 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This file is worth reviewing, independently of the other conversions.

The "big" structs to consider here are:

1. `Producer`, which emits information about retries to Oximeter
1. `RetryHelper`, which callers can use to automate the transaction retry decision-making logic.